packages/vite/src/node/server/http.ts

```diff
@@ -31,8 +29,14 @@ export async function resolveHttpServer(
 }
 
 export async function resolveHttpsConfig(
-  httpsOption: HttpsServerOptions
-): Promise<HttpsServerOptions> {
+  config: ResolvedConfig
+): Promise<HttpsServerOptions | undefined> {
+  if (config.server.https === false || config.server.https == null) {
+    return undefined
+  }
+  
+  const httpsOption =
+    config.server.https === true ? {} : config.server.https
```

Comment:
Seems this change was requested by @nihalgonsalves, but if I read everything correctly it didn't changed any logic

Also IMO it was a bit more readable before :confused: 

I would suggest at least something like this

```suggestion
  if (!config.server.https) return undefined
  
  const httpsOption =
    typeof config.server.https === 'boolean' ? {} : config.server.https
```

In addition I think the `undefined` can be omitted :thinking: so just `return` is used and it automatically returns `undefined`

The `typeof` check makes it more clear that it can be something different than `boolean`. That is helpful if you read code like me in a non IDE like GitHub Web UI

Reply:
I was just trying to be explicit, but it was a nitpick anyway. Either is fine :)